File: src/GZCTF/Modules/TeamLab/Contracts/TeamLabRemoteAccessContracts.cs
using GZCTF.Modules.TeamLab.Domain.Runtime;

namespace GZCTF.Modules.TeamLab.Contracts;

public sealed record CreateTeamLabRemoteSessionModel(string Reason, bool VncConsole = false);

public sealed record TeamLabRemoteAccessAvailabilityModel(
    int AssetId,
    string AssetName,
    TeamLabRemoteProtocol? Protocol,
    bool Available,
    string? UnavailableReason);

public sealed record TeamLabRemoteSessionModel(
    Guid Id,
    Guid RuntimeId,
    int AssetId,
    string AssetName,
    TeamLabRemoteProtocol Protocol,
    TeamLabRemoteSessionStatus Status,
    string Reason,
    DateTimeOffset CreatedAt,
    DateTimeOffset ExpiresAt,
    DateTimeOffset? ConnectedAt,
    DateTimeOffset? EndedAt,
    string? EndReason);

public sealed record TeamLabRemoteConnectModel(string Url, DateTimeOffset ExpiresAt);

public sealed record TeamLabRemoteSessionListItem(
    TeamLabRemoteSessionModel Session, Guid WorkerNodeId, Guid RequestedByUserId);

public sealed record TeamLabRemoteSessionPage(IReadOnlyList<TeamLabRemoteSessionListItem> Items, long? NextCursor);

File: src/GZCTF/ClientApp/src/vnext/features/admin/teamlab/api/teamlabRemoteAccessContracts.ts
export type TeamLabRemoteProtocol = 'containerTerminal' | 'ssh' | 'rdp'
export type TeamLabRemoteSessionStatus = 'creating' | 'ready' | 'connected' | 'ending' | 'ended' | 'failed'

export interface TeamLabRemoteAccessAvailability {
  assetId: number
  assetName: string
  protocol: TeamLabRemoteProtocol | null
  available: boolean
  unavailableReason: string | null
}

export interface TeamLabRemoteSession {
  id: string
  runtimeId: string
  assetId: number
  assetName: string
  protocol: TeamLabRemoteProtocol
  status: TeamLabRemoteSessionStatus
  reason: string
  createdAt: number
  expiresAt: number
  connectedAt: number | null
  endedAt: number | null
  endReason: string | null
}

export interface TeamLabRemoteConnect {
  url: string
  expiresAt: number
}
